setup.py

```diff
@@ -50,7 +50,8 @@
 ]
 
 install_requires = [
-    'mosaicml[libcloud,wandb,mlflow,oci,gcs]>=0.17.2,<0.18',
+    'mosaicml[libcloud,wandb,oci,gcs]>=0.17.2,<0.18',
+    'mlflow>=2.10,<3',
```

Comment:
does this need to be in requires? can we gate mlflow in the hf ckptr?

Reply:
we could move it to optional, but 🤷‍♂️ theres lots of stuff in this list that could be optional. can jira to clean it up more generally if you want. would technically be a breaking change to move mlflow out of the required deps.

Reply:
It isn't correctly in install_requires so it wouldn't be breaking change? not sure i follow.

Reply:
It isn't correctly in install_requires so it wouldn't be breaking change? not sure i follow.